flask/app/variants.py

```diff
@@ -228,17 +227,136 @@ def parse_gene_panel() -> List[Any]:
         app.logger.error("Not all requested genes were found.")
         abort(400, description="Not all requested genes were found.")
 
-    genes = query.all()
+    # genes_list = query.all()
+
+    app.logger.debug("Found %d genes with panel %s", found_genes, ensgs)
 
-    return ensgs
+    return gene_filter, found_genes
+
+
+def parse_region(region: str):
+    """
+    Parses a given list of ranges (eg. "chr1:0500-0509,chr5:0600-0630,chr18:0440-0330").
+    Returns a filter for use against the variant table to find corresponding variants
+    based on gene start and end positions.
+    """
+    CHR, START, END = 0, 1, 2
+    # format check
+    # 1: chromosome, 2: start position, 3: end position
+    region_pattern = re.compile(r"chr([\da-zA-Z]+):([\d]+)-([\d]+)")
+    matches = [region_pattern.match(r) for r in region.split(",")]
+
+    app.logger.info("Requested region: %s", region)
+
+    if not all(match is not None and len(match.groups()) == 3 for match in matches):
+        app.logger.error("Invalid region format: %s", region)
+        abort(400, description="Invalid region format")
+
+    # prepare set of regions
+    region_list = []
+    for i, r in enumerate(set([match.groups() for match in matches])):
```

Comment:
my initial thought of using enumerate was to loop through the items in the list with a counter and modifying the list in place, but this works too! how you currently have it means we can omit the enumerate function along with the `i` counter